oximeter/collector/src/results_sink.rs

```diff
@@ -98,6 +99,41 @@ pub async fn database_inserter(
                     );
                 }
             }
+
+            // Our internal testing rack will be running a ClickHouse cluster
+            // alongside a single-node installation for a while. We want to handle
+            // the case of these two installations running alongside each other, and
+            // oximeter writing to both of them. On our production racks ClickHouse
+            // will only be run on single-node modality, so we want to ignore all
+            // cases where the `ClickhouseClusterNative` service is not available.
+            if let Some(cluster_client) = &cluster_client {
```

Comment:
It looks like we always configure the cluster_client in production, so on every insert this will wait 100ms then timeout and move on. That's probably fine, and in fact I think it may be the right thing to do. The alternative is to query the database for the clickhouse policy setting on every iteration, which may take nearly as long and add some extra code to the system that has the opportunity to break.

I think I prefer to merge this in as is, especially since the single-node insert happens first and the collector only runs every 5 seconds. Does that make sense to you @bnaecker ?

Reply:
Just want to add that since running both installations side by side is only a temporary thing, I wanted to add as little code as possible, and modify as few systems as I could since this code will have to be removed eventually.

Reply:
argghhhh, sorry @bnaecker! I added my comment above but GitHub didn't show me yours that was there first. I just saw Andrew's

> I think 100ms overhead on each insert is not too bad. Is that preferable to spawning a separate insertion task for the cluster?

Would it make much difference?

Reply:
It just separates them, so that writes to one DB don't block the other, whether they fail or succeed. It might be a bigger change than it's worth, so sort of up to you. Thanks!

Reply:
FWIW I attempted add a separate task [here](https://github.com/oxidecomputer/omicron/blob/main/oximeter/collector/src/agent.rs#L129-L139) two days ago, and I started really getting into the weeds and having to modify way too many parts of the system, so I let it go and went for the current solution instead. TBH I think I prefer the least intrusive option.